src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1503 @@ const ActorInit En_Elf_InitVars = {
+    }
+
+    // temp probably fake match
+    unk2C7 = this->unk_2C7;
+    if (unk2C7 > 0) {
+        this->unk_2C7--;
+    }
+
+    if ((this->unk_2C7 == 0) && (globalCtx->csCtx.state != 0)) {
+        this->unk_2C7 = 1;
+    }
+
+    func_80A04D90(this, globalCtx);
+}
+
+void EnElf_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnElf* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    this->actor.shape.rot.y = this->unk_2BC;
+    this->timer++;
+
+    if (this->fairyFlags & FAIRY_FLAG_BIG) {
+        func_80A04D90(this, globalCtx);
+    }
+}
+
+s32 EnElf_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                           struct Actor* thisx, Gfx** gfx) {
+    static Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
+    s32 pad;
+    f32 scale;
+    Vec3f mtxMult;
+    EnElf* this = THIS;
+
+    if (limbIndex == 8) {
+        scale = ((Math_Sins(this->timer * 4096) * 0.1f) + 1.0f) * 0.012f;
+
+        if (this->fairyFlags & FAIRY_FLAG_BIG) {
+            scale *= 2.0f;
+        }
+
+        scale *= (this->actor.scale.x * 124.99999f);
+        Matrix_MultVec3f(&zeroVec, &mtxMult);
+        Matrix_Translate(mtxMult.x, mtxMult.y, mtxMult.z, MTXMODE_NEW);
+        Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    }
+
+    // do not draw wings for big fairies
+    if (this->fairyFlags & FAIRY_FLAG_BIG) {
+        if (limbIndex == 4 || limbIndex == 7 || limbIndex == 11 || limbIndex == 14) {
+            *dList = NULL;
+        }
+    }
+
+    return 0;
+}
+
+void EnElf_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 alphaScale;
+    s32 envAlpha;
+    EnElf* this = THIS;
+    s32 pad1;
+    Gfx* dList;
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A049B4.s")
+    if ((this->unk_2A8 != 8) && !(this->fairyFlags & 8)) {
+        if (!(player->stateFlags1 & 0x100000) || (kREG(90) < this->actor.projectedPos.z)) {
+            dList = Graph_Alloc(globalCtx->state.gfxCtx, sizeof(Gfx) * 4);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A04D90.s")
+            OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_elf.c", 2730);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A04DE4.s")
+            func_80094B58(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A04F94.s")
+            envAlpha = (this->timer * 50) & 0x1FF;
+            envAlpha = (envAlpha > 255) ? 511 - envAlpha : envAlpha;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A05040.s")
+            alphaScale = this->disappearTimer < 0 ? (this->disappearTimer * (7.0f / 6000.0f)) + 1.0f : 1.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Elf/func_80A05114.s")
+            gSPSegment(oGfxCtx->polyXlu.p++, 0x08, dList);
+            gDPPipeSync(dList++);
```

Comment:
```suggestion
            gDPPipeSync(dListHead++);
```

Reply:
this variable in general should be called dListHead? idk why thats confusing me, i thought we use that convention when you have one variable for the whole dlist and one for the front of it

Reply:
I think it makes sense to use the name consistently whether there is another variable or not.

In general tho, the pointer should be treated as the head of a display list if it's iterated over (ie the pointer is incremented). Naming it `dList` implies it's a pointer to the display list itself, which is no longer true once you start iterating over it.